akamai/resource_akamai_property_activation.go

```diff
@@ -366,18 +390,31 @@ func findExistingActivation(property *papi.Property, activation *papi.Activation
 		papi.StatusZone3:               true,
 	}
 	for _, a := range activations.Activations.Items {
+		edge.PrintfCorrelation("[DEBUG]", correlationid, fmt.Sprintf(" SET VERSION UPDATE LOOP TEST loop values %s version %d Activationpassed in  %s version %d  \n", a.Network, a.PropertyVersion, activation.Network, activation.PropertyVersion))
+	}
+	for _, a := range activations.Activations.Items {
+
+		edge.PrintfCorrelation("[DEBUG]", correlationid, " AExisting activation search loop")
+		edge.PrintfCorrelation("[DEBUG]", correlationid, fmt.Sprintf(" AExisting activation search loop inProgressStates %v Activations loop values %s version %d Activationpassed in  %s version %d  \n", inProgressStates[a.Status], a.Network, a.PropertyVersion, activation.Network, activation.PropertyVersion))
+
 		if _, ok := inProgressStates[a.Status]; !ok {
+			edge.PrintfCorrelation("[DEBUG]", correlationid, fmt.Sprintf(" AExisting activation search IF inProgressStates %v Activations loop values %s version %d Activationpassed in  %s version %d  \n", inProgressStates[a.Status], a.Network, a.PropertyVersion, activation.Network, activation.PropertyVersion))
 			continue
 		}
 
 		// There is an activation in progress, if it's for the same version/network/type we can re-use it
-		if a.PropertyVersion != activation.PropertyVersion || a.ActivationType != activation.ActivationType || a.Network != activation.Network {
+		/*if !(a.PropertyVersion != activation.PropertyVersion || a.ActivationType != activation.ActivationType || a.Network != activation.Network) {
 			return nil, fmt.Errorf("%s already in progress: v%d on %s", activation.ActivationType, activation.PropertyVersion, activation.Network)
 		}
-
+		*/
 		//log.Println("[DEBUG] Existing activation found")
-		edge.PrintfCorrelation("[DEBUG]", correlationid, " AExisting activation found")
-		return a, nil
+		if a.PropertyVersion == activation.PropertyVersion && a.ActivationType == activation.ActivationType && a.Network == activation.Network {
```

Comment:
Worth noting that I have confirmed a PAPI data locking issue can occur when both staging and prod requests are received simultaneously and try to update the same lock row.  We are discussing internally and working on a solution to the problem which might be as simple as delaying prod activations so that the deadlock does not occur due to same row simultaneous edit.  But initially it appears that this happens when both PROD and STAGE are activating the same version and update at the same time.  

Not 100% sure this is the full problem space since it is a runtime concurrency problem and those are problematic to test exhaustively.  But it at least gives us a reference for the problem cause on papi side.